presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowPageUtils.java

```diff
@@ -0,0 +1,686 @@
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                long value = vector.get(i);
+                long millis = TimeUnit.SECONDS.toMillis(value);
+                type.writeLong(builder, millis);
+            }
+        }
+        return builder.build();
+    }
+
+    public static Block buildCharTypeBlockFromVarcharVector(VarCharVector vector, Type type)
+    {
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                String value = new String(vector.get(i), StandardCharsets.UTF_8);
+                type.writeSlice(builder, Slices.utf8Slice(CharMatcher.is(' ').trimTrailingFrom(value)));
+            }
+        }
+        return builder.build();
+    }
+
+    public static Block buildTimeTypeBlockFromVarcharVector(VarCharVector vector, Type type)
+    {
+        BlockBuilder builder = type.createBlockBuilder(null, vector.getValueCount());
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                builder.appendNull();
+            }
+            else {
+                String timeString = new String(vector.get(i), StandardCharsets.UTF_8);
+                LocalTime time = LocalTime.parse(timeString);
+                long millis = Duration.between(LocalTime.MIN, time).toMillis();
+                type.writeLong(builder, millis);
+            }
+        }
+        return builder.build();
+    }
+
+    public static Block buildBlockFromListVector(ListVector vector, Type type)
+    {
+        if (!(type instanceof ArrayType)) {
+            throw new IllegalArgumentException("Type must be an ArrayType for ListVector");
+        }
+
+        ArrayType arrayType = (ArrayType) type;
+        Type elementType = arrayType.getElementType();
+        BlockBuilder arrayBuilder = type.createBlockBuilder(null, vector.getValueCount());
+
+        for (int i = 0; i < vector.getValueCount(); i++) {
+            if (vector.isNull(i)) {
+                arrayBuilder.appendNull();
+            }
+            else {
+                BlockBuilder elementBuilder = arrayBuilder.beginBlockEntry();
+                UnionListReader reader = vector.getReader();
+                reader.setPosition(i);
+
+                while (reader.next()) {
+                    Object value = reader.readObject();
+                    if (value == null) {
+                        elementBuilder.appendNull();
+                    }
+                    else {
+                        appendValueToBuilder(elementType, elementBuilder, value);
+                    }
+                }
+                arrayBuilder.closeEntry();
```

Comment:
what does this do?

Reply:
closeEntry() ensures that the system knows where one array ends and the next begins.